Index: Editor.pm
===================================================================
--- Editor.pm	(revision 14491)
+++ Editor.pm	(working copy)
@@ -11,13 +11,14 @@
 use Padre::Wx                 ();
 use Padre::Wx::FileDropTarget ();
 use Padre::Wx::Role::Main     ();
+use Wx::Scintilla             ();
 use Padre::Logger;
 
 our $VERSION    = '0.85';
 our $COMPATIBLE = '0.81';
 our @ISA        = qw{
 	Padre::Wx::Role::Main
-	Wx::StyledTextCtrl
+	Wx::ScintillaTextCtrl
 };
 
 # Convenience colour constants
